lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5216,6 +5232,36 @@ void ConformanceChecker::checkConformance(MissingWitnessDiagnosisKind Kind) {
       }
     }
   }
+
+  if (Conformance->isInvalid()) {
+    return;
+  }
+
+  // As a narrow fix for a source compatibility issue with SwiftUI's
+  // swiftinterface, but only if the conformance succeeds, warn about an
+  // actually malformed conformance if we recorded a 'typealias' type witness
+  // with an underlying type of 'Never', which resides in a context that does
+  // not satisfy the requirements of the conformance context.
+  //
+  // FIXME: If SwiftUI redeclares the typealias under the correct constraints,
+  // this can be removed.
+  if (InvalidSwiftUITypeWitnessAndFailedReq) {
+    auto *assocTypeDecl = std::get<0>(*InvalidSwiftUITypeWitnessAndFailedReq);
```

Comment:
I personally find std::tuple awkward for this reason. Can you use a bespoke struct, even if it's only used once, just to name the fields?